problem-spring-web-autoconfigure/pom.xml

```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.zalando</groupId>
+        <artifactId>problem-spring-parent</artifactId>
+        <version>0.25.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>problem-spring-web-autoconfigure</artifactId>
+
+    <name>Problem: Spring Web Autoconfiguration</name>
+    <description>Spring autoconfiguration module for Problem Spring Web</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure-processor</artifactId>
+            <version>${spring-boot.version}</version>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>${spring.version}</version>
+            <optional>true</optional>
```

Comment:
Why optional?

Reply:
My understanding was that everything in the autonconfigure module should be optional so as not to force transitive dependencies for someone using the autoconfigure module and that it was only in the starter module where you forced a set of specific dependencies; but correct me if i'm wrong. i've never done spring autoconfigure modules before

Reply:
I believe Spring is doing that because they bundled all auto configurations into a single jar. This isn't needed if you package an auto configuration into a designated artifact which we would be doing here.

Reply:
are you talking specifically about _all_ spring dependencies here or only this one in particular?
The only ones that i'm sure that should optional are the ones to your own libraries that you are autoconfiguring

> You should mark the dependencies to the library as optional so that you can include the autoconfigure module in your projects more easily. If you do it that way, the library is not provided and, by default, Spring Boot backs off.

Understanding why they recommend that is where I (mis)took that everything should be optional and only force if you use the starter

Reply:
also, I found this presentation about spring autoconfig and this was the sample code
https://github.com/snicoll-demos/spring-boot-master-auto-configuration/blob/master/hornetq-spring-boot-autoconfigure/pom.xml

Reply:
The idea behind that is that your autoconfiguration module has no hard dependency on your own library (or rather on the library that it autoconfigures). By doing that you allow to have it on the classpath without transitively pulling in your library. 

Not exactly sure what the purpose would be since, there is no obvious reason to pull the autoconfigure module if you don't intent to use it. Anyway, the dependency on spring parts can be made non-optional.

Reply:
i thought the same that you did. My own rationalization to that recommendation is that if maybe you want to use the autoconfigure module without the starter, it means that you are in mode "i will control what gets pulled", and in that mode you don't want the autoconfigure module of each library pulling their own version of spring. I know maven will converge all those conflicts anyway, but i figured that's why all dependencies in the autoconfig pom should be optional.

anyway, i left the `optional` only in the zalando dependencies, in `spring-boot-autoconfigure-processor` (since the documentation explicitly says so) and in the spring-security dependencies (since you don't want to pull security if you're not pulling it already in your own project)